tests/test_lattice.py

```diff
@@ -0,0 +1,883 @@
+        gracefully without crashing.
+        """
+        # Arrange: Create an empty CustomizeLattice instance.
+        empty_lattice = CustomizeLattice(
+            dimensionality=2, identifiers=[], coordinates=[]
+        )
+
+        # Assert: Verify basic properties.
+        assert empty_lattice.num_sites == 0
+        assert len(empty_lattice) == 0
+
+        # Act & Assert for .show(): Verify it prints the expected message without crashing.
+        caplog.set_level(logging.INFO)
+
+        empty_lattice.show()
+        assert "Lattice is empty, nothing to show." in caplog.text
+
+        # Act & Assert for neighbor finding: Verify these calls run without errors.
+        empty_lattice._build_neighbors()
+        assert empty_lattice.get_neighbor_pairs(k=1) == []
+
+    def test_single_site_lattice_handles_gracefully(self):
+        """
+        Tests that a lattice with a single site correctly handles neighbor
+        finding (i.e., returns no neighbors).
+        """
+        # Arrange: Create a CustomizeLattice with a single site.
+        single_site_lattice = CustomizeLattice(
+            dimensionality=2, identifiers=[0], coordinates=[[0.0, 0.0]]
+        )
+
+        # Assert: Verify basic properties.
+        assert single_site_lattice.num_sites == 1
+
+        # Act: Attempt to build neighbor relationships.
+        single_site_lattice._build_neighbors(max_k=1)
+
+        # Assert: The single site should have no neighbors.
+        assert single_site_lattice.get_neighbors(0, k=1) == []
+
+    def test_access_with_non_existent_identifier(self, simple_square_lattice):
+        """
+        Tests that a ValueError is raised when accessing a site
+        with an identifier that does not exist in the lattice.
+        """
+        # Arrange
+        lattice = simple_square_lattice
+        invalid_identifier = "non_existent_site"
+
+        # Act & Assert
+        # Your code raises a ValueError with a specific message. We can even
+        # use the 'match' parameter to check if the error message is correct.
+        with pytest.raises(ValueError, match="not found in the lattice"):
+            lattice.get_index(invalid_identifier)
+
+        with pytest.raises(ValueError, match="not found in the lattice"):
+            lattice.get_site_info(invalid_identifier)
+
+    def test_show_warning_for_unsupported_dimension(self, caplog):
+        """
+        Tests that .show() prints a warning when called on a lattice with a
+        dimensionality that it does not support for plotting (e.g., 4D).
+        """
+        # Arrange: Create a simple lattice with an unsupported dimension.
+        lattice_4d = CustomizeLattice(
+            dimensionality=4, identifiers=[0], coordinates=[[0, 0, 0, 0]]
+        )
+
+        # Act
+        lattice_4d.show()
+
+        # Assert: Check that the appropriate warning was printed to stdout.
+        assert "show() is not implemented for 4D lattices." in caplog.text
+
```

Comment:
I suggest to add one or few tests on relative large lattice, say $6 \times 8$, and to check whether neighbor searching is correctly handled for large `max_k`, say 4, 6, 8 and more, what if the given `max_k` is larger than the possible `max_k`?